src/transformers/models/seggpt/image_processing_seggpt.py

```diff
@@ -0,0 +1,627 @@
+                Number of classes in the segmentation task (excluding the background). If specified, a palette will be
+                built, assuming that class_idx 0 is the background, to map the prompt mask from a single class_idx
+                channel to a 3 channel RGB. Not specifying this will result in the prompt mask either being passed
+                through as is if it is already in RGB format or being duplicated across the channel dimension.
+        """
+        do_resize = do_resize if do_resize is not None else self.do_resize
+        do_rescale = do_rescale if do_rescale is not None else self.do_rescale
+        do_normalize = do_normalize if do_normalize is not None else self.do_normalize
+        resample = resample if resample is not None else self.resample
+        rescale_factor = rescale_factor if rescale_factor is not None else self.rescale_factor
+        image_mean = image_mean if image_mean is not None else self.image_mean
+        image_std = image_std if image_std is not None else self.image_std
+
+        size = size if size is not None else self.size
+        size_dict = get_size_dict(size)
+
+        images = make_list_of_images(images)
+
+        if not valid_images(images):
+            raise ValueError(
+                "Invalid image type. Must be of type PIL.Image.Image, numpy.ndarray, "
+                "torch.Tensor, tf.Tensor or jax.ndarray."
+            )
+
+        if do_resize and size is None:
+            raise ValueError("Size must be specified if do_resize is True.")
+
+        if do_rescale and rescale_factor is None:
+            raise ValueError("Rescale factor must be specified if do_rescale is True.")
+
+        if do_normalize and (image_mean is None or image_std is None):
+            raise ValueError("Image mean and std must be specified if do_normalize is True.")
+
+        # All transformations expect numpy arrays.
+        images = [to_numpy_array(image) for image in images]
+
+        if is_scaled_image(images[0]) and do_rescale:
+            logger.warning_once(
+                "It looks like you are trying to rescale already rescaled images. If the input"
+                " images have pixel values between 0 and 1, set `do_rescale=False` to avoid rescaling them again."
+            )
+
+        if input_data_format is None:
+            # We assume that all images have the same channel dimension format.
+            # If mask is PIL could have 2 or 3 dimensions, hence the if-else check
+            input_data_format = (
+                infer_channel_dimension_format(images[0]) if images[0].ndim > 2 else ChannelDimension.LAST
+            )
+
+        if is_mask:
+            palette = self.get_palette(num_labels) if num_labels is not None else None
+            # Since this is the input for the next steps and transformation
+            images = [self.mask_to_rgb(image, palette, input_data_format=input_data_format) for image in images]
```

Comment:
if `images[0].ndim <= 2` it doesn't make sense to say `ChannelDimension.LAST` here. There's no dimension at all. 

I'm assuming this is to account for the masks? In this case I'd do something like the following: 

```suggestion
        if input_data_format is None and not is_mask:
            # We assume that all images have the same channel dimension format.
            input_data_format = infer_channel_dimension_format(images[0])
        
        if is_mask:
            palette = self.get_palette(num_labels) if num_labels is not None else None
            # Since this is the input for the next steps and transformation
            images = [self.mask_to_rgb(image, palette, data_format=ChannelDimension.FIRST) for image in images]
            input_data_format = ChannelDimension.FIRST
```

Reply:
https://github.com/huggingface/transformers/pull/27735/commits/3373cbb63e5c3b364027b54b2bbcfbd4433f9d25